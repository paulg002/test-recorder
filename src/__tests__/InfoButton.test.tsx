import * as React from 'react';
import { shallow } from 'enzyme';
import InfoButton from '../popup/components/InfoButton';
import '../setupTests';

describe('InfoButton', () => {
  let handleInfoToggleMock;
  let wrapper;
  let props;
  beforeAll(() => {
    handleInfoToggleMock = jest.fn();
    props = {
      toggleInfoDisplay: handleInfoToggleMock,
    };
  });
  it('Button text should display "Info" when shouldInfoDisplay is false', () => {
    props.shouldInfoDisplay = false;
    wrapper = shallow((<InfoButton {...props} />));
    expect(wrapper).toMatchSnapshot();
    const button = wrapper.find('button');
    expect(button.text()).toBe('Info');
  });
  it('Button text should display "Recording Menu" when shouldInfoDisplay is true', () => {
    props.shouldInfoDisplay = true;
    wrapper = shallow((<InfoButton {...props} />));
    const button = wrapper.find('button');
    expect(button.text()).toBe('Recording Menu');
  });
  it('Should invoke toggleInfoDisplay when button is clicked', () => {
    props.shouldInfoDisplay = true;
    wrapper = shallow((<InfoButton {...props} />));
    const button = wrapper.find('button');
    button.simulate('click');
    expect(handleInfoToggleMock).toHaveBeenCalled();
  });
});
