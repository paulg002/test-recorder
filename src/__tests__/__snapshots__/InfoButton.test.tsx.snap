// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoButton Button text should display "Info" when shouldInfoDisplay is false 1`] = `
<Fragment>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Info
  </button>
</Fragment>
`;
